frontend/src/app/components/modal/modal.component.ts

```diff
@@ -0,0 +1,37 @@
+import { Component, ViewChild, EventEmitter, Input, Output, OnChanges, SimpleChanges, TemplateRef } from "@angular/core";
+import { ModalService } from "src/app/components/modal/service/modal.service";
+
+@Component({
+  selector: 'modal',
+  templateUrl: './modal.component.html'
+})
+export class ModalComponent implements OnChanges {
+  @ViewChild('customTemplate') customTemplate!: TemplateRef<any>;
+  @Input() open = false;
+  @Output() openChange = new EventEmitter<boolean>;
+
+  constructor(
+    private modalService: ModalService
+  ) {}
+
```

Comment:
this is the main logic that open/close modal depending on the INPUT open that the parent change.

It could be more simple but no time